src/SourceBrowser.Site/Repositories/BrowserRepository.cs

```diff
@@ -32,6 +33,52 @@ internal static string GetDocumentHtml(string username, string repository, strin
             }
         }
 
+        internal static bool TryLockRepository(string userName, string repoName)
+        {
+	        string lockFileDirectory = Path.Combine(StaticHtmlAbsolutePath, userName, repoName);
+            string lockFilePath = Path.Combine(lockFileDirectory, Constants.REPO_LOCK_FILENAME);
+	        if (File.Exists(lockFilePath))
+	        {
+		        return false;
+	        }
+	        else
+	        {
+		        // Create a file that indicates that the upload will begin
+		        Directory.CreateDirectory(lockFileDirectory);
+                File.WriteAllText(lockFilePath, Constants.REPO_STATUS_PROCESSING);
+		        return true;
+	        }
+        }
+
+        internal static void UnlockRepository(string userName, string repoName)
+        {
+	        string lockFilePath = Path.Combine(StaticHtmlAbsolutePath, userName, repoName, Constants.REPO_LOCK_FILENAME);
+	        if (File.Exists(lockFilePath))
+	        {
+		        File.Delete(lockFilePath);
```

Comment:
Absolutely. Will locking a static object suffice?
`static Object fileLock = new Object();
...
lock (fileLock)
{`
I'm not overly concerned about using only one lock even though we could have as many locks as there are repositories. The object is locked only for the duration of writing to or removing a file, which is very short compared to the entire upload process. Opinions?